install/kubernetes/helm/istio/values.yaml

```diff
@@ -299,6 +299,24 @@ mixer:
   replicaCount: 1
   image: mixer
 
+  istio-policy:
+    autoscaleEnabled: true
+    autoscaleMin: 1
+    autoscaleMax: 5
+    cpu:
+      targetAverageUtilization: 80
+    memory:
+      targetAverageValue: 2500Mi
```

Comment:
If the question is what is the application doing that uses memory, I don't have a definitive breakdown. For istio-telemetry, a large amount of memory can be spent on handling (deserializing, etc.) the attribute bag per request that it receives in batches of 1000 (with the need to handle multiple batches concurrently).

TBH, I'm not sure what an appropriate value is for this target. I was hoping to get some feedback from the perf team, etc. However, earlier issues said memory usage was spiking up to 20G+ with ~5k RPS at ingress (which means more total requests following through the mesh). My intent is that this value would lead to scaling well before that explosion.

Reply:
I have some concern for this, currently, the Kubenretes build-in policy only works with CPU utilization for HPA, and if you want to enable some other metrics such as memory etc, you may want to install some other components https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#how-does-the-horizontal-pod-autoscaler-work , I can see most customers may not have this installed, so not sure if the HPA will report error if it cannot handle the memory utilization as metrics?

Reply:
Is that true?

> Notice that you can specify other resource metrics besides CPU. By default, the only other supported resource metric is memory.

https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/

Reply:
we can stick with CPU for 1.0
Memory is trickier anyways.